hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/exports/FileSignTool.java

```diff
@@ -0,0 +1,478 @@
+import com.swirlds.common.crypto.HashingOutputStream;
+import com.swirlds.common.crypto.SignatureType;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.stream.EventStreamType;
+import com.swirlds.common.stream.StreamType;
+import com.swirlds.common.stream.internal.StreamTypeFromJson;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.MarkerManager;
+import org.apache.logging.log4j.core.LoggerContext;
+
+/**
+ * This is a standalone utility tool to generate signature files for event/record stream, and
+ * account balance files generated by stream server.
+ *
+ * <p>It can also be used to sign any single files with any extension. For files except .evts and
+ * .rcd, the Hash in the signature file is a SHA384 hash of all bytes in the file to be signed.
+ *
+ * <p>For .evts files, it generates version 5 signature files.
+ *
+ * <p>For .rcd files, it generates version 6 signature files.
+ *
+ * <p>Please see README.md for format details
+ */
+public class FileSignTool {
+    public static final String CSV_EXTENSION = ".csv";
+    public static final String ACCOUNT_BALANCE_EXTENSION = ".pb";
+    public static final String SIG_FILE_NAME_END = "_sig";
+    /** next bytes are signature */
+    public static final byte TYPE_SIGNATURE = 3;
+    /** next 48 bytes are hash384 of content of the file to be signed */
+    public static final byte TYPE_FILE_HASH = 4;
+
+    public static final String STREAM_TYPE_JSON_PROPERTY = "streamTypeJson";
+    public static final String LOG_CONFIG_PROPERTY = "logConfig";
+    public static final String FILE_NAME_PROPERTY = "fileName";
+    public static final String KEY_PROPERTY = "key";
+    public static final String DEST_DIR_PROPERTY = "destDir";
+    public static final String ALIAS_PROPERTY = "alias";
+    public static final String PASSWORD_PROPERTY = "password";
+    public static final String DIR_PROPERTY = "dir";
```

Comment:
nit: These fields if not being shared should be private

Reply:
updated